src/troute-network/troute/HYFeaturesNetwork.py

```diff
@@ -13,71 +13,96 @@
 from datetime import datetime
 from pprint import pformat
 import os
-
+import fiona
 import troute.nhd_io as nhd_io #FIXME
 from troute.nhd_network import reverse_dict, extract_connections, reverse_network, reachable
 from .rfc_lake_gage_crosswalk import get_rfc_lake_gage_crosswalk, get_great_lakes_climatology
-
+import re
 __verbose__ = False
 __showtiming__ = False
 
+def find_layer_name(layers, pattern):
+    """
+    Find a layer name in the list of layers that matches the regex pattern.
+    """
+    for layer in layers:
+        if re.search(pattern, layer, re.IGNORECASE):
+            return layer
+    return None
+
 def read_geopkg(file_path, compute_parameters, waterbody_parameters, cpu_pool):
-    # Establish which layers we will read. We'll always need the flowpath tables
-    layers = ['flowpaths','flowpath_attributes']
-
-    # If waterbodies are being simulated, read lakes table
-    if waterbody_parameters.get('break_network_at_waterbodies',False):
-        layers.append('lakes')
-        layers.append('nexus')
-
-    # If any DA is activated, read network table as well for gage information
-    data_assimilation_parameters = compute_parameters.get('data_assimilation_parameters',{})
-    streamflow_nudging = data_assimilation_parameters.get('streamflow_da',{}).get('streamflow_nudging',False)
-    usgs_da = data_assimilation_parameters.get('reservoir_da',{}).get('reservoir_persistence_usgs',False)
-    usace_da = data_assimilation_parameters.get('reservoir_da',{}).get('reservoir_persistence_usace',False)
-    rfc_da = data_assimilation_parameters.get('reservoir_da',{}).get('reservoir_rfc_da',{}).get('reservoir_rfc_forecasts',False)
-    if any([streamflow_nudging, usgs_da, usace_da, rfc_da]):
-        layers.append('network')
+    # Retrieve available layers from the GeoPackage
+    available_layers = fiona.listlayers(file_path)
+
+    # patterns for the layers we want to find
+    layer_patterns = {
+        'flowpaths': r'flow[-_]?paths?',
+        'flowpath_attributes': r'flow[-_]?path[-_]?attributes?',
```

Comment:
I think we need to drop the `"paths"` here. I believe these can be called `"flowlines"` which this would not find.

Reply:
I think if we drop the "paths" and make it to just look for "flow" it will find both "flowpaths" layer and "flowpath_attribute" layer. I just added "flowlines" too, that should solve the problem

Reply:
Great. Can you do the same for flowline-attributes? I'm not sure if that's a potential table name, but we might as well cover our bases.

Reply:
Sure. Just added